### DISCUSSION TITLE
CORS issue trying to bundle markdown-language-features into vscode for web

### DISCUSSION THREAD

**Discussion by jlewi** [microsoft/vscode-discussions#985] [Extension Development QnA] [answered]
HI 

I'm building vscode for web and serving it on my own domain. As part of this, I'm trying to bundle in the [markdown-language-features](https://github.com/microsoft/vscode/tree/09623a546e4c7ac980723aea520414d0921f4768/extensions/markdown-language-features) extension. 

When I serve markdown-language-features on my development server (localhost); I get the following CORS error.

```
Access to fetch at 'http://localhost:8080/extensions/markdown-language-features/notebook-out/index.js' from origin 'https://1erdbq6hogdhabs4gm0ohn07kj92ahpgeu0g88jtkdfamn9c5i64.vscode-cdn.net' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

I'm confused by this error. Why is it downloading and running scripts from vscode-cnd.net that are trying to talk to my server?

**Comment by jlewi** [comment]
I updated the CORS policy on my server and that fixed it but I'd love to understand what's happening here. In particular what assets are being pulled from vscode-cdn.net?

**Comment by tamuratak** [comment, marked as answer]
> what assets are being pulled from vscode-cdn.net?

The contents of WebViews, the HTML file and script files.

For security reasons, WebView must be isolated from the host of vscode. WebView must have a different origin from the host. So, webview contents are hosted on `*.vscode-cdn.net` for VS Code Web. 

See

- https://github.com/microsoft/vscode/blob/b48a213e0e1623f27cb7476ad4d5f8768b00e04d/src/vs/workbench/services/environment/browser/environmentService.ts#L227-L237 
- https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy
